Show as of dates for key toxicities in narrative

gregquinn2001 opened the pull request:
Now key toxicities show as of dates in narrative view.

_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._


## Submitter:

**Running the Application**

- [X] Manually tested in Chrome
- [ ] Manually tested in IE

**Documentation**

- [X] This pull request describes why these changes were made
- [X] Recognizes any potential shortcomings/bugs in the description 

**Code Quality**

- [X] 4-space indents - convert any tabs to spaces
- [X] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [X] Code is commented

**Tests**

- [X] Existing tests passed
- [ ] Added Enzyme-UI tests for slim mode 
- [ ] Added Enzyme-UI tests for full mode
- [ ] Added backend tests for new functionality added
- [ ] Added note parser examples to test new functionality


## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code


## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

jafeltra commented:
I noticed that in narrative mode, the as of dates on toxicities show up as structured data and let you insert them into a note, but in the tabular view, they are plain text. Should they both either be plain text or structured data? Or is there a reason that they are different?

gregquinn2001 commented:
Yeah I should have noted that in my pull request description. I agree it's not ideal. I could easily remove the dates as structured in narrative but I'm not sure how difficult it would be to make the dates be a separate structured value in tabular. Should I just remove the structured in narrative so they are consistent and create a jira or just create a jira or do I have to get the tabular structured dates to work before you'll approve!?

jafeltra commented:
I think you can just create a jira. If the goal is to eventually have both be structured data and it isn't simple to get tabular structured dates to work now, I'd say it's okay to have us halfway to the goal, rather than go backwards.

nicoleng12 commented:
[NN1] When you switch to narrative mode and try to insert the toxicity date into a note, it inserts the grade. Given that (and looking at the comments above), for this PR maybe just remove the date as structured in narrative and make a jira for adding the toxicity dates as structured in both narrative and tabular.

sgober commented:
test comment

sgober commented:
test comment 2

Dtphelan1 commented:
test take two?

sgober commented:
testing again

sgober commented:
test

Dtphelan1 commented:
more specific test?

Dtphelan1 commented:
last time?

gregquinn2001 commented:
fixed NN1 plus another bug I found in demo1. plus did some further cleanup of data formats and transforms in summaries. as of dates are now just text in both tabular and narrative modes.

Also rebased.

nicoleng12 commented:
This looks good! NN1 has been addressed. Backend tests pass
